pkg/kube/util.go

```diff
@@ -12,6 +12,55 @@ import (
 	k8yaml "k8s.io/apimachinery/pkg/util/yaml"
 )
 
+var placeholder, _ = regexp.Compile(`(?mU)<(.*)>`)
+
+// replaceableInner recurses through the given map and returns true if _any_ value is a `<placeholder>` string
+func replaceableInner(node *map[string]interface{}) bool {
+	obj := *node
+
+	for _, value := range obj {
+		valueType := reflect.ValueOf(value).Kind()
+		if valueType == reflect.Map {
+			inner, ok := value.(map[string]interface{})
+			if !ok {
+				panic(fmt.Sprintf("Deserialized YAML node is non map[string]interface{}"))
```

Comment:
Are we using `panic` now instead of returning an error back to the generate method?

Reply:
No, that is what the current code does in `replaceInner` as well. I copied most of that to create this method. We can change it if you want, but since it's non-recoverable I just left it as-is

Reply:
As long as we are consistent with the error handling. I'm also not a fan of printing out the stack trace and i believe panic will do that. Maybe not in argocd as the error does not get logged but as a cli tool

Reply:
> I'm also not a fan of printing out the stack trace and i believe panic will do that

That's right, it does print the stack trace. But honestly in these kinds of scenarios, if the plugin fails because of this problem I think the stack trace would help debug the problem. For now I say we keep it, and maybe leave it as a pre 1.0 item to replace with returning `error`'s instead

Reply:
Will fix as part of: https://github.com/IBM/argocd-vault-plugin/issues/63